sklearn/tree/_criterion.pyx

```diff
@@ -291,6 +326,18 @@ cdef class ClassificationCriterion(Criterion):
         self.weighted_n_samples = weighted_n_samples
         self.weighted_n_node_samples = 0.0
 
+        self.pointer_reset(start, end)
+
+        return 0
+
+    cdef int pointer_reset(
+        self,
+        SIZE_t start,
+        SIZE_t end
+    ) nogil except -1:
```

Comment:
where is `self.start` and `self.end` being assigned to the new `start` and `end` values?

Reply:
Thank you! I'll make sure these get assigned to instance variables